src/cmd/services/m3coordinator/downsample/downsampler_test.go

```diff
@@ -2841,6 +2842,53 @@ func TestDownsamplerWithOverrideNamespace(t *testing.T) {
 	testDownsamplerAggregation(t, testDownsampler)
 }
 
+func TestSafeguardInProcessDownsampler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store := kv.NewMockStore(ctrl)
+	store.EXPECT().SetIfNotExists(gomock.Eq(matcher.NewOptions().NamespacesKey()), gomock.Any()).Return(0, nil).Times(1)
+	store.EXPECT().Set(gomock.Any(), gomock.Any()).Times(0)
+	store.EXPECT().CheckAndSet(gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+	store.EXPECT().Delete(gomock.Any()).Times(0)
+	_ = newTestDownsampler(t, testDownsamplerOptions{
+		remoteClientMock: nil,
+		kvStore:          store,
+	})
+}
+
+func TestDownsamplerNamespacesEtcdInit(t *testing.T) {
+	t.Run("does not reset namespaces key", func(t *testing.T) {
+		store := mem.NewStore()
+		initialNamespaces := rulepb.Namespaces{Namespaces: []*rulepb.Namespace{{Name: "testNamespace"}}}
+		_, err := store.Set(matcher.NewOptions().NamespacesKey(), &initialNamespaces)
+		require.NoError(t, err)
+
+		_ = newTestDownsampler(t, testDownsamplerOptions{kvStore: store})
+
+		assert.Equal(t, initialNamespaces, readNamespacesKey(t, store), 1)
+	})
+
+	t.Run("initializes namespace key", func(t *testing.T) {
+		store := mem.NewStore()
+
+		_ = newTestDownsampler(t, testDownsamplerOptions{kvStore: store})
+
+		ns := readNamespacesKey(t, store)
+		require.NotNil(t, ns)
+		assert.Len(t, ns.Namespaces, 0)
+	})
+
+	t.Run("do not initialize namespaces when RequireNamespaceWatchOnInit is true", func(t *testing.T) {
```

Comment:
Nit for consistency:
```suggestion
	t.Run("does not initialize namespaces when RequireNamespaceWatchOnInit is true", func(t *testing.T) {
```